plugins/services/src/js/pages/EditFrameworkConfiguration.js

```diff
@@ -0,0 +1,143 @@
+import React, { PropTypes } from "react";
+import mixin from "reactjs-mixin";
+import { StoreMixin } from "mesosphere-shared-reactjs";
+import { routerShape } from "react-router";
+import deepEqual from "deep-equal";
+import CosmosPackagesStore from "#SRC/js/stores/CosmosPackagesStore";
+import UniversePackage from "#SRC/js/structs/UniversePackage";
+import FrameworkConfiguration from "#SRC/js/components/FrameworkConfiguration";
+import Loader from "#SRC/js/components/Loader";
+import RequestErrorMsg from "#SRC/js/components/RequestErrorMsg";
+
+class EditFrameworkConfiguration extends mixin(StoreMixin) {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      packageDetails: null,
+      deployErrors: null,
+      formData: null,
+      formErrors: {},
+      hasError: false
+    };
+
+    this.store_listeners = [
+      {
+        name: "cosmosPackages",
+        events: [
+          "serviceDescriptionError",
+          "serviceDescriptionSuccess",
+          "serviceUpdateSuccess",
+          "serviceUpdateError"
+        ]
+      }
+    ];
+
+    CosmosPackagesStore.fetchServiceDescription(props.params.id);
+  }
+
+  onCosmosPackagesStoreServiceDescriptionSuccess() {
+    const fullPackage = CosmosPackagesStore.getServiceDetails();
+    const packageDetails = new UniversePackage(fullPackage.package);
+    const formData = this.reorderResolvedOptions(
+      fullPackage.resolvedOptions,
+      packageDetails.getConfig()
+    );
+
+    this.setState({ packageDetails, formData });
+  }
+
+  onCosmosPackagesStoreServiceDescriptionError() {
+    this.setState({ hasError: true });
+  }
+
+  onCosmosPackagesStoreServiceUpdateSuccess() {
+    this.handleGoBack();
+  }
+
+  onCosmosPackagesStoreServiceUpdateError(deployErrors) {
+    this.setState({ deployErrors });
+  }
+
+  // necessary to have review screen same order of tabs as the form
```

Comment:
🎨 I'd move the comment to https://github.com/dcos/dcos-ui/pull/2443/files#diff-efd3f92415ec5298ecb20a1ff4642c0eR42 as it's not document the functionality but the why it needs to be applied.